sqllineage/core/parser/sqlfluff/extractors/dml_insert_extractor.py

```diff
@@ -97,9 +99,15 @@ def extract(
                     conditional_handler.handle(segment, holder)
 
         # By recursively extracting each subquery of the parent and merge, we're doing Depth-first search
+
+        # we are passing the write context as the subquery for type SUPPORTED_STMT_TYPES (defined above)
+        # would always be a source(read) query and it is important to pass
+        # the write context to generate the correct lineage
         for sq in subqueries:
+            holder.add_read(sq)
```

Comment:
added the subquery as read in holder

Reply:
For top-level subquery in DML like: 1) create table foo as (subquery); 2) insert into foo (subquery), I think the best approach is just to not treat them as subquery by stripping the bracket directly. The code change is pushed.